llvm/include/llvm/ADT/DenseMap.h

```diff
@@ -226,6 +226,12 @@ class DenseMapBase : public DebugEpochBase {
     return Default;
   }
 
+  ValueT lookup_or(const_arg_type_t<KeyT> Val, const ValueT &Default) const {
```

Comment:
Sorry, I may have misguided you in an earlier PR of yours.  Would you mind trying something like this?  I've taken this from `std::optional::value_or`.

```suggestion
  template <typename U = std::remove_cv_t<ValueT> >
  ValueT lookup_or(const_arg_type_t<KeyT> Val, U &&Default) const {
```

Reply:
I'm not sure how to test this, as I don't have a Windows machine, and the Windows CI isn't configured the same way as the Buildbot that failed. Wasn't the error `C:\b\slave\clang-x64-windows-msvc\llvm-project\llvm\lib\Analysis\ScalarEvolution.cpp(15907): note: You cannot bind an lvalue to an rvalue reference`? If I understand correctly, MSVC has its own version of STL, so what you proposed might not work.

Reply:
Perhaps you can try a mock-up version of `DenseMap::lookup_or` in Compiler Explorer or something!?  After that, I'd rely on the CI and buildbots.

The `template <typename U = std::remove_cv_t<ValueT> >` trick should be part of the C++ standard, not specific to a specific implementation of the standard.

Reply:
With `U` being a template parameter, this is no longer an rvalue reference but a [universal/forwarding reference](https://www.cppreference.com/w/cpp/language/reference.html#Forwarding_references). Such a reference can bind to lvalues, in which case `U` will be derived as lvalue reference type using [reference collapsing](https://www.cppreference.com/w/cpp/language/reference.html#Reference_collapsing). So this should work.

To be honest, I have no idea what the default argument does, since it seems like this should always be deduced.